Added the "What is Gulp" section, including .feature2 and highlightjs

ansballard opened the pull request:
Threw together the "What is Gulp" section from the "Getting Started" mockup (https://github.com/gulpjs/gulpjs.github.io/issues/38). The existing page didn't include the headers on each section, so I didn't add it to this one either. I also added highlightjs, since that's what was used on the previous version of the site. Tried my best to follow the styles already in `index.html` and `main.css`, but I'm happy to change anything that doesn't fit. And if you all would like all the sections implemented for the issue I'll tack onto this pull request for the other two. Just ran out of time today.

contra commented:
Can you provide a screenshot of the page before and after these changes?

phated commented:
Hmmm, I think highlightjs is overkill.  Could you copy the styling out of the page and just add it directly into the page?

contra commented:
Also seems like a lot of this copy overlaps with stuff already on the page. If it seems like we already conveyed the info then maybe there is no need to add this section.

ansballard commented:
1. Will add before/after screenshots when I get home
1. Will do, I actually tried that first but I didn't want to reimplement something that might get used somewhere else
1. I agree, most of it's pretty generic. I do like the contrast and the opportunity to show a simple example of a gulpfile/tasks though. Maybe it just needs different points, or maybe it should focus on something else entirely?

Will update when 1 and 2 are changed.

ansballard commented:
Swapped the code example for an example jade build task based on API.md. Also added `gulpfile.js` in the header of the example code box, in case it's not 100% clear that this is a gulpfile instead of using the API from a node script or something like that. Kept the commits separate in case you all don't like that though.

![gulp-replacedexamplecode](https://cloud.githubusercontent.com/assets/3535127/21743623/b06bf36c-d4cb-11e6-813c-cb0099715832.png)

contra commented:
How about this for a code sample?

```js
var gulp = require('gulp');
var jade = require('gulp-jade');
var less = require('gulp-less');
var minifyCSS = require('gulp-csso');

gulp.task('html', function(){
  return gulp.src('client/templates/*.jade')
    .pipe(jade())
    .pipe(gulp.dest('build/html'))
});

gulp.task('css', function(){
  return gulp.src('client/templates/*.less')
    .pipe(less())
    .pipe(minifyCSS())
    .pipe(gulp.dest('build/css'))
});

gulp.task('default', [ 'html', 'css' ]);
```

I'd be okay losing the copy entirely and just doing a big code sample IMO, I think we have enough copy on the page.

ansballard commented:
Removed the copy, along with the css I added for it (`.list` and `.listitem`). Also removed `feature2`, since the code example is a single `.col-12`. Also moved the section to `.container-sml` since it doesn't take up very much horizontal space, and didn't do anything extra for small screens since it seems to shrink down fine. Also fixed a couple tabs from the last couple commits where I didn't double-indent.

![gulp-nocopy](https://cloud.githubusercontent.com/assets/3535127/21747109/f6f144c6-d521-11e6-97fd-f95da1d696ce.png)

contra commented on css/main.css:
```diff
@@ -301,6 +301,65 @@ p.paragraph {
+    color: rgba(255,255,255,0.9);
+}
+.whatisgulp .listitem-heading {
+    color: rgba(255,255,255,0.9);
```
2 space padding is the format

contra commented on index.html:
```diff
@@ -79,6 +79,42 @@ <h3 class="subheading center">Quality Ecosystem</h3>
   </div>
 
+  <div class="whatisgulp">
+    <div class="container-sml">
```
Larger container - theres too much empty space around the editor so it looks a little funky

contra commented on css/main.css:
```diff
@@ -301,6 +301,65 @@ p.paragraph {
+}
+.editor-window code {
+    background-color: #141414;
+    font: normal 10pt monospace;
```
Use the font family and font-size from the CLI example above.

ansballard replied:
so I copied out the font family (Menlo, etc...) and the font-size (40px) but since it doesn't scale like the svg, it doesn't really look right. Also `gulpfile.js` and the rest of the header look way too small in comparison now, should they be scaled up as well?

![gulp-clifonts](https://cloud.githubusercontent.com/assets/3535127/21751652/645b0a92-d590-11e6-809e-7b86ca1f26c6.png)

ansballard replied:
Example of how `cli.svg` could look using `<pre><code`.

[Examples with resizing, ~20mb gifs](http://imgur.com/a/Fmy0J)

contra replied:
@ansballard It looks like highlightjs is doing something weird with the font-family on strings - doesn't seem to be monospace.

Your call on svg vs. pre, btw. Can also use breakpoints to adjust font size so it looks alright on mobile and desktop.

ansballard commented:
Oof, sorry for the dirty merging. I replaced the CLI example at the top of the page with `<pre><code>` and used the same class as the code example so it should scale the same, have a similar feel, etc. Also added breakpoints to the code example on small screens, and scaled the font size down from 14pt to 13pt so it fits a little better. As far as I can tell all the comments above are addressed, I'll list them to be sure.

1. Highlightjs font issue: Fixed, needed more specific css for those spans
1. Replace cli.svg with <pre><code> and add breakpoints to code example: done
1. Font sizes have fit the page fairly well
1. Code function indented
1. Indentation looks like it matches the rest of the file to me
1. Switched the code example to a large container, which looks fine with the larger font

contra commented:
@ansballard Screenshots?

ansballard commented:
Sorry about that, totally forgot.

CLI Fullscreen

![gulp-clilarge](https://cloud.githubusercontent.com/assets/3535127/21831153/6285fc6c-d767-11e6-9b20-139ee06a7f7c.png)

CLI Small Screen

![gulp-clismall](https://cloud.githubusercontent.com/assets/3535127/21831158/6a6caf16-d767-11e6-98e1-89a9324255fc.png)

Code Example Fullscreen

![gulp-codeexamplelarge](https://cloud.githubusercontent.com/assets/3535127/21831164/7289d912-d767-11e6-89b6-619f0e58e914.png)

Code Example Small Screen

![gulp-codeexamplesmall](https://cloud.githubusercontent.com/assets/3535127/21831171/78d340e2-d767-11e6-9593-fd4a7d73036a.png)

contra commented:
Thanks!
